sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -119,7 +119,7 @@ abstract class DataType extends AbstractDataType {
 
   override private[sql] def acceptsType(other: DataType): Boolean = sameType(other)
 
-  def physicalDataType: PhysicalDataType = UninitializedPhysicalType
+  private[sql] def physicalDataType: PhysicalDataType = UninitializedPhysicalType
```

Comment:
can we change the modifier in subclasses as well?

Reply:
done